compiler/circle-input-names/src/circle-input-names.cpp

```diff
@@ -0,0 +1,181 @@
+template <> struct is_VariadicOut<CircleWhile> : std::true_type {};
+// clang-format on
+
+// For Circle Nodes which have variadic arity and variadic outputs
+template <typename CircleOp, typename std::enable_if_t<is_VariadicArity<CircleOp>::value &&
+                                                       is_VariadicOut<CircleOp>::value> * = nullptr>
+auto CircleNodeCreator()
+{
+  return CircleOp(1, 1);
+}
+
+// For Circle Nodes which have variadic arity but single output
+template <typename CircleOp,
+          typename std::enable_if_t<is_VariadicArity<CircleOp>::value &&
+                                    !is_VariadicOut<CircleOp>::value> * = nullptr>
+auto CircleNodeCreator()
+{
+  return CircleOp(1);
+}
+
+// For Circle Nodes which have fixed arity
+template <typename CircleOp,
+          typename std::enable_if_t<!is_VariadicArity<CircleOp>::value> * = nullptr>
+auto CircleNodeCreator()
+{
+  return CircleOp();
+}
+
+// Add fused activation function option to CircleNode if it supports FusedActFunc traits
+void add_fused_actfn_option(CircleNode *node)
+{
+  auto node_ = dynamic_cast<CircleNodeMixin<CircleNodeTrait::FusedActFunc> *>(node);
+  if (node_)
+  {
+    node_->fusedActivationFunction(luci::FusedActFunc::RELU);
+  }
+}
+
+// Add padding option to AVERAGE_POOL_2D, CONV_2D, DEPTHWISE_CONV_2D, L2_POOL_2D, MAX_POOL_2D,
+// TRANSPOSE_CONV nodes
+void add_padding_option(CircleNode *node)
+{
+  switch (node->opcode())
+  {
+#define CIRCLE_NODE(OPCODE, CLASS)            \
+  case luci::CircleOpcode::OPCODE:            \
+  {                                           \
+    auto node_ = dynamic_cast<CLASS *>(node); \
+    node_->padding(Padding::SAME);            \
+    break;                                    \
+  }
+    CIRCLE_NODE(AVERAGE_POOL_2D, CircleAveragePool2D)
+    CIRCLE_NODE(CONV_2D, CircleConv2D)
+    CIRCLE_NODE(DEPTHWISE_CONV_2D, CircleDepthwiseConv2D)
+    CIRCLE_NODE(L2_POOL_2D, CircleL2Pool2D)
+    CIRCLE_NODE(MAX_POOL_2D, CircleMaxPool2D)
+    CIRCLE_NODE(TRANSPOSE_CONV, CircleTransposeConv)
+#undef CIRCLE_NODE
+    default:
+    {
+      break;
+    }
+  }
+  return;
+}
+
+// Add mode option to MIRROR_PAD, ROPE nodes
+void add_mode_option(CircleNode *node)
+{
+  switch (node->opcode())
+  {
+#define CIRCLE_NODE(OPCODE, CLASS)            \
+  case luci::CircleOpcode::OPCODE:            \
+  {                                           \
+    auto node_ = dynamic_cast<CLASS *>(node); \
+    auto mode_ = node_->mode();               \
+    node_->mode(decltype(mode_)(1));          \
+    break;                                    \
+  }
+    CIRCLE_NODE(MIRROR_PAD, CircleMirrorPad)
+    CIRCLE_NODE(ROPE, CircleRoPE)
+#undef CIRCLE_NODE
+    default:
+    {
+      break;
+    }
+  }
+  return;
+}
+
+// Mock Symbol Table for CircleNodeSummaryBuilder
+class MockSymbolTable : public locop::SymbolTable
+{
+  std::string lookup(const loco::Node *) const override { return ""; }
+};
+
```

Comment:
### Note

In fact, this tool is nothing but prints all input names defined in `CircleNodeSummaryBuilder`.
However, it requires some hacks to instantiate the builder outside of the module.
All above codes are the **Hacks**.